runtime-modules/council/src/benchmarking.rs

```diff
@@ -0,0 +1,756 @@
+#![cfg(feature = "runtime-benchmarks")]
+use super::*;
+use frame_benchmarking::{account, benchmarks, Zero};
+use frame_support::traits::{OnFinalize, OnInitialize};
+use frame_system::EventRecord;
+use frame_system::Module as System;
+use frame_system::RawOrigin;
+use membership::Module as Membership;
+use sp_runtime::traits::{Bounded, One};
+use sp_std::convert::TryInto;
+use sp_std::prelude::*;
+use sp_std::vec;
+use sp_std::vec::Vec;
+
+use crate::Module as Council;
+
+const SEED: u32 = 0;
+
+// We create this trait because we need to be compatible with the runtime
+// in the mock for tests. In that case we need to be able to have `membership_id == account_id`
+// We can't create an account from an `u32` or from a memberhsip_dd,
+// so this trait allows us to get an account id from an u32, in the case of `64` which is what
+// the mock use we get the parameter as a return.
+// In the case of `AccountId32` we use the method provided by `frame_benchmarking` to get an
+// AccountId.
+pub trait CreateAccountId {
+    fn create_account_id(id: u32) -> Self;
+}
+
+impl CreateAccountId for u64 {
+    fn create_account_id(id: u32) -> Self {
+        id.into()
+    }
+}
+
+impl CreateAccountId for sp_core::crypto::AccountId32 {
+    fn create_account_id(id: u32) -> Self {
+        account::<Self>("default", id, SEED)
+    }
+}
+
+fn assert_last_event<T: Trait>(generic_event: <T as Trait>::Event) {
+    if !cfg!(test) {
```

Comment:
why do we need such checks in benchmarking?

Reply:
This was due to an old bug in the council pallet where events were not generated in the mock runtime. I'll get rid of this right away.